pkg/apiserver/debugapi/endpoint_def.go

```diff
@@ -0,0 +1,517 @@
+						{Value: "miss-peer"},
+						{Value: "extra-peer"},
+						{Value: "down-peer"},
+						{Value: "pending-peer"},
+						{Value: "offline-peer"},
+						{Value: "empty-peer"},
+						{Value: "hist-peer"},
+						{Value: "hist-keys"},
+					}),
+					Name: "state", Required: true},
+			},
+			QueryParams: []*endpoint.APIParam{
+				{Model: endpoint.APIParamModelInt, Name: "bound"},
+			},
+			OnReceiveRequest: func(req *endpoint.Request) error {
+				state := req.PathValues.Get("state")
+				val := req.QueryValues.Get("bound")
+
+				if val == "" {
+					if strings.EqualFold(state, "hist-size") {
+						req.QueryValues.Set("bound", "10")
+					} else if strings.EqualFold(state, "hist-keys") {
+						req.QueryValues.Set("bound", "10000")
+					}
+				}
+				return nil
+			},
+		}).
+		AddEndpoint(&endpoint.APIModel{
+			ID:        "pd_scheduler_show",
+			Component: model.NodeKindPD,
+			Path:      "/schedulers",
+			Method:    endpoint.MethodGet,
+			QueryParams: []*endpoint.APIParam{
+				{Model: endpoint.APIParamModelText, Name: "status"},
+			},
+		}).
+		AddEndpoint(&endpoint.APIModel{
+			ID:        "pd_stores",
+			Component: model.NodeKindPD,
+			Path:      "/stores",
+			Method:    endpoint.MethodGet,
+			QueryParams: []*endpoint.APIParam{
+				{Model: endpoint.APIParamModelTags, Name: "state"},
+			},
+			OnReceiveRequest: func(req *endpoint.Request) error {
+				vals := req.QueryValues["state"]
+				req.QueryValues.Del("state")
+				if len(vals) != 0 {
+					for _, state := range vals {
+						stateValue, ok := metapb.StoreState_value[state]
+						if !ok {
+							return fmt.Errorf("unknown state: %s", state)
+						}
+						req.QueryValues.Add("state", strconv.Itoa(int(stateValue)))
+					}
+				}
+				return nil
+			},
+		}).
+		AddEndpoint(&endpoint.APIModel{
+			ID:        "pd_store_id",
+			Component: model.NodeKindPD,
+			Path:      "/store/{storeID}",
+			Method:    endpoint.MethodGet,
+			PathParams: []*endpoint.APIParam{
+				{Model: endpoint.APIParamModelText, Name: "storeID", Required: true},
+			},
+		}).
+		AddEndpoint(&endpoint.APIModel{
+			ID:        "pd_pprof",
+			Component: model.NodeKindPD,
+			Path:      "/debug/pprof/{kind}",
+			Method:    endpoint.MethodGet,
+			PathParams: []*endpoint.APIParam{
+				pprofKindsParam,
+			},
+			QueryParams: []*endpoint.APIParam{
+				{Model: endpoint.APIParamModelConstant("1"), Name: "debug"},
+				pprofSecondsParam,
+			},
+			OnReceiveRequest: func(req *endpoint.Request) error {
+				return timeoutMiddleware(req, req.QueryValues.Get("seconds"))
+			},
+		})
+
+	// tikv
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "tikv_config",
+		Component: model.NodeKindTiKV,
+		Path:      "/config",
+		Method:    endpoint.MethodGet,
+	}).
+		AddEndpoint(&endpoint.APIModel{
+			ID:        "tikv_profile",
+			Component: model.NodeKindTiKV,
+			Path:      "/debug/pprof/profile",
+			Method:    endpoint.MethodGet,
+			Ext:       ".svg",
```

Comment:
Is it possible to use mimeTypes to handle it? From the architecture point of view it is wired for an endpoint to have a n "Ext" field. This is very coupled with the business.

Reply:
The endpoint doesn't return the correct mime type for now.
I have an idea to implement the "true middleware" with `next`. Then we got a AOP struct. We can manipulate response mime type in custom middleware after the "send request middleware" exec `next`. That would be better.